packages/core/src/controllers/ConfigCtrl.ts

```diff
@@ -23,14 +27,13 @@ export const ConfigCtrl = {
     return valtioSub(state, () => callback(state))
   },
 
-  setConfig(config: ConfigOptions) {
-    if (!config.projectId)
-      throw new Error(
-        'Web3Modal requires projectId that can be obtained at cloud.walletconnect.com'
-      )
+  setConfig(config: ConfigCtrlState) {
+    OptionsCtrl.setStandaloneChains(config.standaloneChains)
+    OptionsCtrl.setIsStandalone(Boolean(config.standaloneChains?.length))
+    OptionsCtrl.setIsCustomMobile(Boolean(config.mobileWallets?.length))
+    OptionsCtrl.setIsCustomDesktop(Boolean(config.desktopWallets?.length))
+    OptionsCtrl.setIsExplorer(Boolean(config.projectId?.length))
```

Comment:
Minor: TS Pickup: Unnecessary optional chain of a non nullish value. `config.projectId` resolves

Reply:
`projectId` became optional, so can be undefined here. Did you see this error in your editor? Maybe run `yarn` to update everything so that new types are picked up.

Reply:
My bad, resolved!